tests/e2e/policy_driven_vol_allocation.go

```diff
@@ -1374,6 +1377,635 @@ var _ = ginkgo.Describe("[vol-allocation] Policy driven volume space allocation
+		pvcs2d := [][]*v1.PersistentVolumeClaim{}
+
+		rand.New(rand.NewSource(time.Now().UnixNano()))
+		suffix := fmt.Sprintf("-%v-%v", time.Now().UnixNano(), rand.Intn(10000))
+		categoryName := "category" + suffix
+		tagName := "tag" + suffix
+
+		catID, tagID := createCategoryNTag(ctx, categoryName, tagName)
+		defer func() {
+			deleteCategoryNTag(ctx, catID, tagID)
+		}()
+
+		attachTagToDS(ctx, tagID, sharedvmfsURL)
+		defer func() {
+			detachTagFromDS(ctx, tagID, sharedvmfsURL)
+		}()
+
+		allocationTypes := []string{
+			thinAllocType,
+			lztAllocType,
+		}
+
+		ginkgo.By("create 2 SPBM policies with thin, LZT volume allocation respectively")
+		ginkgo.By("Create 2 SCs each with a SPBM policy created from step 1")
+		ginkgo.By("Create two PVCs of 10g using each of the SCs created from step 2")
+		for _, at := range allocationTypes {
+			policyID, policyName := createVmfsStoragePolicy(
+				ctx, pc, at, map[string]string{categoryName: tagName})
+			defer func() {
+				deleteStoragePolicy(ctx, pc, policyID)
+			}()
+			scParameters[scParamStoragePolicyName] = policyName
+			policyNames = append(policyNames, policyName)
+			policyIds = append(policyIds, policyID)
+			storageclass, pvclaim, err := createPVCAndStorageClass(client,
+				namespace, nil, scParameters, "10Gi", nil, "", false, "")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			pvcs = append(pvcs, pvclaim)
+			scs = append(scs, storageclass)
+			pvclaim2, err := createPVC(client, namespace, nil, "10Gi", storageclass, "")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			pvcs = append(pvcs, pvclaim2)
+			pvcs2d = append(pvcs2d, []*v1.PersistentVolumeClaim{pvclaim})
+		}
+
+		defer func() {
+			ginkgo.By("Delete the SCs created in step 2")
+			for _, sc := range scs {
+				err := client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		}()
+
+		ginkgo.By("Verify the PVCs created in step 3 are bound")
+		pvs, err := fpv.WaitForPVClaimBoundPhase(client, pvcs, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		volIds := []string{}
+		ginkgo.By("Verify that the created CNS volumes are compliant and have correct policy id")
+		for i, pv := range pvs {
+			volumeID := pv.Spec.CSI.VolumeHandle
+			storagePolicyMatches, err := e2eVSphere.VerifySpbmPolicyOfVolume(volumeID, policyNames[i/2])
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(storagePolicyMatches).To(gomega.BeTrue(), "storage policy verification failed")
+			e2eVSphere.verifyVolumeCompliance(volumeID, true)
+			e2eVSphere.verifyDatastoreMatch(volumeID, []string{sharedvmfsURL})
+			volIds = append(volIds, volumeID)
+		}
+
+		defer func() {
+			ginkgo.By("Delete the PVCs created in step 3")
+			for _, pvc := range pvcs {
+				err := fpv.DeletePersistentVolumeClaim(client, pvc.Name, namespace)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
```

Comment:
I think we can cover this is in a single for loop as we did for other testcases

Reply:
I found that in one of my runs some time back mappings had gone wonky so had them separated.